modules/cloud-provider-equinix-metal/templates/metal-cloud-config.yaml.tftpl

```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: metal-cloud-config
+  namespace: kube-system
+stringData:
+  cloud-sa.json: |
+    ${jsonencode(cpem_secret)}
```

Comment:
One thing I'm confused about is the `cloud-config` parameter for CPEM.  The CPEM README indicates that the `cloud-config` option is (1) specified in the secret and (2) specifies where the secret is, so I'm not clear how that works.  Does CPEM look at 2 secrets if the `cloud-config` parameter is specified?

https://github.com/equinix/cloud-provider-equinix-metal#configuration

Reply:
The `cloud-config` argument is inherited from Kubenernetes itself where controllers used to have built-in functionality for working with certain clouds.  That functionality has since been moved to external cloud provider ([cloud controller managers](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)).

https://github.com/kubernetes/kubernetes/search?p=1&q=cloud-config

We continue to use that parameter because our provider inherits it here: https://github.com/kubernetes/cloud-provider/blob/5a236942a3ca6f0454bfcb979cc31207af4894b2/options/cloudprovider.go#L35-L42

This is not to be confused with CloudInit / CloudConfig - there is no relation.